packages/blocks/src/api/parser.js

```diff
@@ -395,29 +395,14 @@ export function getMigratedBlock( block, parsedAttributes ) {
 }
 
 /**
- * Creates a block with fallback to the unknown type handler.
- *
- * @param {Object} blockNode Parsed block node.
+ * Convert derivative blocks to their canonical form. This function is used
+ * both in the parser level for previous content and to convert such blocks
+ * used in Custom Post Types templates.
  *
- * @return {?Object} An initialized block object (if possible).
+ * @param {string} name
+ * @param {Object} attributes
  */
-export function createBlockWithFallback( blockNode ) {
-	const { blockName: originalName } = blockNode;
-	let { attrs: attributes, innerBlocks = [], innerHTML } = blockNode;
-	const { innerContent } = blockNode;
-	const freeformContentFallbackBlock = getFreeformContentHandlerName();
-	const unregisteredFallbackBlock =
-		getUnregisteredTypeHandlerName() || freeformContentFallbackBlock;
-
-	attributes = attributes || {};
-
-	// Trim content to avoid creation of intermediary freeform segments.
-	innerHTML = innerHTML.trim();
-
-	// Use type from block content if available. Otherwise, default to the
-	// freeform content fallback.
-	let name = originalName || freeformContentFallbackBlock;
-
+export function convertDerivativeBlocks( name, attributes ) {
```

Comment:
It's not a very important detail, but I wonder if `convertOldBlocks` — or something along those lines — might be a better name for this. I consider that only a few cases, like Embeds, are about converting derivatives. The other ones, like Text, are about converting old forms of these blocks.

Reply:
Yeah, the name didn't seem right to me, but wanted to get first feedback about the approach. I'll change it.

Reply:
Does `convertLegacyBlocks` sound alright?

Reply:
Yep!